dotnet/src/Connectors/VectorData.Abstractions/ConnectorSupport/VectorStoreRecordModelBuilder.cs

```diff
@@ -295,7 +295,9 @@ private void SetPropertyStorageName(VectorStoreRecordPropertyModel property, str
             return;
         }
 
-        property.StorageName = storageName;
+        property.StorageName = this.Options.EscapeIdentifier is not null
```

Comment:
this is important and I want to rely on it in other PRs: it's really hard to find all the usages of `StorageName` to do proper escaping. Since we can now perform the escaping during model build, we can ensure that whatever `StorageName` returns, it's safe to be consumed.  Moreover, it's done only once (perf).